src/main/java/org/terasology/climateConditions/BodyTemperatureSystem.java

```diff
@@ -0,0 +1,143 @@
+    private float criticalLowBodyTemperatureThreshold = 0.1f;
+    private float lowBodyTemperatureThreshold = 0.2f;
+    private float reducedBodyTemperatureThreshold = 0.3f;
+    private float raisedBodyTemperatureThreshold = 0.5f;
+    private float highBodyTemperatureThreshold = 0.6f;
+    private float criticalHighBodyTemperatureThreshold = 0.7f;
+    //The Normal Body Temperature range is 0.3 - 0.5 as of now.
+
+    public void postBegin() {
+        boolean processedOnce = false;
+        for (EntityRef entity : entityManager.getEntitiesWith(WorldComponent.class)) {
+            if (!processedOnce) {
+                delayManager.addPeriodicAction(entity, BODY_TEMPERATURE_UPDATE_ACTION_ID, 0, CHECK_INTERVAL);
+                processedOnce = true;
+            } else {
+                logger.warn("More than one entity with WorldComponent found");
+            }
+        }
+        // The world is the entity here which shall have a Body Temperature Update Periodic Action.
+    }
+
+    @ReceiveEvent
+    public void onTemperatureUpdate(PeriodicActionTriggeredEvent event, EntityRef world) {
+        if (event.getActionId().equals(BODY_TEMPERATURE_UPDATE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(AliveCharacterComponent.class,
+                    BodyTemperatureComponent.class, LocationComponent.class)) {
+                LocationComponent location = entity.getComponent(LocationComponent.class);
+                BodyTemperatureComponent bodyTemperature = entity.getComponent(BodyTemperatureComponent.class);
+                float envTemperature = climateConditionsSystem.getTemperature(location.getLocalPosition());
+                float envHumidity = climateConditionsSystem.getHumidity(location.getLocalPosition());
+                float deltaTemp =
+                        ((((envTemperature - (envHumidity / 10)) - bodyTemperature.current) / 200000) * CHECK_INTERVAL);
+
+                //Send event for other systems to modify change in body temperature.
+                AffectBodyTemperatureEvent affectBodyTemperatureEvent = new AffectBodyTemperatureEvent(deltaTemp);
+                entity.send(affectBodyTemperatureEvent);
+                deltaTemp = affectBodyTemperatureEvent.getResultValue();
+
+                //Check for change in body temperature levels.
+                float oldValue = bodyTemperature.current;
+                //Update current body temperature.
+                bodyTemperature.current = bodyTemperature.current + deltaTemp;
+                float newValue = bodyTemperature.current;
+                entity.saveComponent(bodyTemperature);
+                if (oldValue != newValue) {
+                    entity.send(new BodyTemperatureValueChangedEvent(oldValue, newValue));
+                }
+
+                //only for development purposes
+                entity.getOwner().send(new ChatMessageEvent("Body Temperature: " + bodyTemperature.current,
```

Comment:
Hah, here we have the chat messages to be replaced with some icon placeholder in the UI 🤓 😃 different PR, though 😉

Reply:
Yeah, about that... if this is removed, it will be very hard to change body temperatures according to one's will in order to see all the effects during the play test...
Currently the env. temperatures are not very uniformly distributed according to biomes..  And if this is removed, a separate UI for env. temperature would also be required and while for displaying body temperature, using a UI wouldn't be a big problem as it won't have drastic transitions, the env. temp just keeps changing  drastically.... so that won't be very good.... so i think.. It would still require the numeric value,  otherwise color would just keep changing....
I'm not sure.. what do you think?

Reply:
Hm, we could use a color gradient like you used for the filling icon thing in the first iteration... this would probably lead to funny color changes, but in the end it's just supposed to be a bit nicer than the plain chat message ^^